### DISCUSSION TITLE
Total Solar Power Generated per Day

### DISCUSSION THREAD

**Discussion by JasonMM2000** [vwt12eh8/hassio-ecoflow#40] [Ideas]
Hello,

I'm always curious how much energy is actually generated per day from my solar arrays.  Right now I have two 180w panels hooked up for testing and it would be great to see how much energy they produced per day.  

Please let me know if that is something do-able.  

Thank you very much for putting this together!

**Comment by vwt12eh8** [comment]
Couldn't you solve this by registering the "MPPT input energy" entity in the energy dashboard?

**Reply by JasonMM2000** [reply]
The stat that would need to be added up over the day is the Delta Pro DC Input.  
![DeltaProDCOutput](https://user-images.githubusercontent.com/111086259/184419154-49535b82-d871-4e6b-9bba-0b48f67c0813.JPG)

Thanks for taking a look at this

**Reply by vwt12eh8** [reply]
If you want to calculate the energy using DC input entities, you can use the "Integration - Riemann sum integral" helper.
For integral method, we recommend "left".

https://www.home-assistant.io/integrations/integration
